src/index.js

```diff
@@ -2028,144 +2303,21 @@ const initialize = async () => {
-    // Wait until warning rendered or not to improve accuracy
-    if (!scrollToHandled) {
-      handleScrollTo({ delay: true });
-    }
-  }
-
-  function handleEIP1559Support(supported) {
-    if (supported && Array.isArray(accounts) && accounts.length >= 1) {
-      sendEIP1559Button.disabled = false;
-      sendEIP1559Button.hidden = false;
-      sendButton.innerText = 'Send Legacy Transaction';
-    } else {
-      sendEIP1559Button.disabled = true;
-      sendEIP1559Button.hidden = true;
-      sendButton.innerText = 'Send';
-    }
-  }
-
-  function handleNewNetwork(networkId) {
-    networkDiv.innerHTML = networkId;
-  }
-
-  async function handleScrollTo({ delay = false } = {}) {
-    if (!scrollTo) {
-      return;
-    }
-
-    scrollToHandled = true;
-
-    console.log('Attempting to scroll to element with ID:', scrollTo);
-
-    const scrollToElement = document.getElementById(scrollTo);
-
-    if (!scrollToElement) {
-      console.warn('Cannot find element with ID:', scrollTo);
-      return;
-    }
-
-    if (delay) {
-      await new Promise((resolve) => setTimeout(resolve, 1000));
-    }
-
-    scrollToElement.scrollIntoView();
-  }
-
-  async function getNetworkAndChainId() {
-    try {
-      const chainId = await ethereum.request({
-        method: 'eth_chainId',
-      });
-      handleNewChain(chainId);
-
-      const networkId = await ethereum.request({
-        method: 'net_version',
-      });
-      handleNewNetwork(networkId);
-
-      const block = await ethereum.request({
-        method: 'eth_getBlockByNumber',
-        params: ['latest', false],
-      });
-
-      handleEIP1559Support(block.baseFeePerGas !== undefined);
-    } catch (err) {
-      console.error(err);
-    }
-  }
-
-  updateButtons();
+  /**
+   * Providers
+   */
 
-  if (isMetaMaskInstalled()) {
-    ethereum.autoRefreshOnNetworkChange = false;
-    getNetworkAndChainId();
+  useWindowProviderButton.onclick = setActiveProviderDetailWindowEthereum;
+};
 
-    ethereum.on('chainChanged', () => getNetworkAndChainId());
-    // networkChanged is deprecated, but there is no other way to ensure we catch every network ID change
-    ethereum.on('networkChanged', () => getNetworkAndChainId());
-    ethereum.on('accountsChanged', (newAccounts) => {
-      ethereum
-        .request({
-          method: 'eth_getBlockByNumber',
-          params: ['latest', false],
-        })
-        .then((block) => {
-          handleEIP1559Support(block.baseFeePerGas !== undefined);
-        });
-      handleNewAccounts(newAccounts);
-    });
+/**
+ * Entrypoint
+ */
 
-    try {
-      const newAccounts = await ethereum.request({
-        method: 'eth_accounts',
-      });
-      handleNewAccounts(newAccounts);
-    } catch (err) {
-      console.error('Error on init when getting accounts', err);
-    }
-  } else {
-    handleScrollTo();
-  }
+const initialize = async () => {
+  initializeFormElements();
+  setActiveProviderDetailWindowEthereum();
+  detecEip6963();
```

Comment:
detect

Reply:
Resolved in [6744431](https://github.com/MetaMask/test-dapp/pull/263/commits/67444312011c83ce2785a3929ffaa0d7302d2e22)